src/RA_AchievementOverlay.cpp

```diff
@@ -724,74 +719,80 @@ void AchievementOverlay::DrawAchievementExaminePage(HDC hDC, int nDX, _UNUSED in
             nDots = 0;
 
         const int nDotCount = nDots / 25;
-        sprintf_s(buffer, 256, " Loading.%c%c%c ", nDotCount >= 1 ? '.' : ' ', nDotCount >= 2 ? '.' : ' ',
-                  nDotCount >= 3 ? '.' : ' ');
-
-        TextOut(hDC, nDX + nLoadingMessageX, nLoadingMessageY, NativeStr(buffer).c_str(), strlen(buffer));
+        // we need to reallocate buffer or heap corruption will happen
+        buffer = std::string(13, char());
+        // TODO: ra::StringPrintf needs a handler for "%c"
+        // Size is 13 but capacity is larger
+        Ensures(sprintf_s(buffer.data(), buffer.capacity(), " Loading.%c%c%c ", nDotCount >= 1 ? '.' : ' ',
+                          nDotCount >= 2 ? '.' : ' ', nDotCount >= 3 ? '.' : ' ') >= 0);
+        TextOut(hDC, nDX + nLoadingMessageX, nLoadingMessageY, NativeStr(buffer).c_str(),
+                gsl::narrow<int>(buffer.length()));
     }
 }
 
 void AchievementOverlay::DrawNewsPage(HDC hDC, int nDX, _UNUSED int, const RECT& rcTarget) const
 {
-    unsigned int nYOffset = 90;
+    auto nYOffset = 90;
 
-    const unsigned int nHeaderGap = 4;
-    const unsigned int nArticleGap = 10;
+    constexpr auto nHeaderGap = 4;
+    constexpr auto nArticleGap = 10;
 
-    const unsigned int nBorder = 32;
+    constexpr auto nBorder = 32;
 
-    const unsigned int nLeftAlign = nDX + nBorder;
-    const unsigned int nRightAlign = nDX + (rcTarget.right - nBorder);
+    const auto nLeftAlign = nDX + nBorder;
+    const auto nRightAlign = nDX + (rcTarget.right - nBorder);
 
-    const unsigned int nArticleIndent = 20;
+    constexpr auto nArticleIndent = 20;
 
-    const int nWidth = rcTarget.right - rcTarget.left;
-    const int nHeight = rcTarget.bottom - rcTarget.top;
+    const auto nWidth = rcTarget.right - rcTarget.left;
+    const auto nHeight = rcTarget.bottom - rcTarget.top;
 
     HGDIOBJ hOldObject = SelectObject(hDC, g_hFontDesc2);
 
-    for (int i = m_nNewsSelectedItem; i < static_cast<int>(m_LatestNews.size()); ++i)
+    for (auto it = std::next(m_LatestNews.begin(), m_nNewsSelectedItem); it != m_LatestNews.end(); ++it)
     {
-        const char* sTitle = m_LatestNews[i].m_sTitle.c_str();
-        const char* sPayload = m_LatestNews[i].m_sPayload.c_str();
+        gsl::cstring_span<> sTitle{it->m_sTitle};
+        gsl::cstring_span<> sPayload{it->m_sPayload};
```

Comment:
Recommend using well-known types wherever possible. Suggest
```
const auto& sTitle = m_LatestNews[i].m_sTitle;
```
and
```
NativeStr(sTitle).c_str()
```
on line 763.

Reply:
Sure